api/modules/files/controller.js

```diff
@@ -1,27 +1,59 @@
 var Boom = require('boom');
 var Promise = require('bluebird');
-var File = require('./model.js');
+
+var File = require('./model');
+var Project = require('../projects/model');
 
 module.exports = {
   getFiles: function(req, reply) {
-    return reply(File.fetchAll());
+    File.fetchAll()
+    .then(function(records) {
+      if(records.models.length < 1) {
+        return reply(Boom.notFound());
+      }
+
+      reply(records.toJSON());
+    })
+    .catch(function(e) {
+      reply(Boom.badRequest(e));
+    });
   },
   getFile: function(req, reply) {
-    return reply(File.query({
+    File.query({
       where: {
         id: req.params.id
       }
-    }).fetch());
+    }).fetch()
+    .then(function(record) {
+      if (!record) {
+        return Promise.reject('File reference does not exist.');
+      }
+      reply(record.toJSON());
+    })
+    .catch(function(e) {
+      reply(Boom.notFound(e));
+    });
   },
   getProjectFiles: function(req, reply) {
-    File.query({
+    Project.query({
       where: {
-        project_id: req.params.project_id
+        id: req.params.project_id
       }
-    }).fetchAll()
+    }).fetch()
+    .then(function(record) {
+      if (!record) {
+        return Promise.reject('Project reference does not exist.');
+      }
+
+      return File.query({
```

Comment:
By returning this promise, the `.then` and `.catch` on lines 54 and 61 will do the trick.